Broadcast properly `cell_measures` when using `extract_shape` with `decomposed: True`

sloosvel opened the pull request:
<!--
    Thank you for contributing to our project!

    Please do not delete this text completely, but read the text below and keep
    items that seem relevant. If in doubt, just keep everything and add your
    own text at the top, a reviewer will update the checklist for you.

-->

## Description

<!--
    Please describe your changes here, especially focusing on why this pull
    request makes ESMValCore better and what problem it solves.

    Before you start, please read our contribution guidelines: https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html

    Please fill in the GitHub issue that is closed by this pull request,
    e.g. Closes #1903
-->
This PR makes sure that time-dependent cell measures are broadcasted properly to a cube with a `shape_id` coordinate.
Closes #2347 

Link to documentation:

***

## [Before you get started](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#getting-started)

- [x] [☝ Create an issue](https://github.com/ESMValGroup/ESMValCore/issues) to discuss what you are going to do

## [Checklist](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#checklist-for-pull-requests)

It is the responsibility of the author to make sure the pull request is ready to review. The icons indicate whether the item will be subject to the [🛠 Technical][1] or [🧪 Scientific][2] review.

<!-- The next two lines turn the 🛠 and 🧪 below into hyperlinks -->
[1]: https://docs.esmvaltool.org/en/latest/community/review.html#technical-review
[2]: https://docs.esmvaltool.org/en/latest/community/review.html#scientific-review

- [x] [🧪][2] The new functionality is [relevant and scientifically sound](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#scientific-relevance)
- [x] [🛠][1] This pull request has a [descriptive title and labels](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#pull-request-title-and-label)
- [x] [🛠][1] Code is written according to the [code quality guidelines](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#code-quality)
- [x] [🧪][2] and [🛠][1] [Documentation](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#documentation) is available
- [x] [🛠][1] [Unit tests](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#tests) have been added
- [x] [🛠][1] Changes are [backward compatible](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#backward-compatibility)
- [x] [🛠][1] Any changed [dependencies have been added or removed](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#dependencies) correctly
- [x] [🛠][1] The [list of authors](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#list-of-authors) is up to date
- [x] [🛠][1] All [checks below this pull request](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#pull-request-checks) were successful

***

To help with the number pull requests:

- 🙏 We kindly ask you to [review](https://docs.esmvaltool.org/en/latest/community/review.html#review-of-pull-requests) two other [open pull requests](https://github.com/ESMValGroup/ESMValCore/pulls) in this repository

schlunma reviewed (commented):
Thanks Saskia, looks good to me, just have two minor comments.

sloosvel commented:
Thanks for the comments @schlunma , I added a comment and changed a bit the condition to also make it work when only one region is extracted and the result cube has no `shape_id` dimension. Here is a recipe to test both cases:

```yaml
  description: |
    This a recipe to monitor experiments by automatizing the creation of plots
    from raw preprocessor output.
  authors:
    - vegas-regidor_javier
  maintainer:
    - loosveldt-tomas_saskia


datasets:
- dataset: EC-Earth3-CC
  ensemble: r1i1p1f1
  grid: gn
  exp: historical
  mip: Omon
  project: CMIP6

preprocessors:
  extract_two_regions:
    extract_shape:
      shapefile: AR6
      method: contains
      crop: False
      decomposed: True
      ids: {Acronym: [EAO, WAF]}
    volume_statistics:
      operator: mean
  extract_one_region:
    extract_shape:
      shapefile: AR6
      method: contains
      crop: False
      decomposed: True
      ids: {Acronym: [EAO]}
    volume_statistics:
      operator: mean

diagnostics:
  timeseries:
    variables:
      thetao_1:
        short_name: thetao
        timerange: '1950/1952'
        preprocessor: extract_two_regions
      thetao_2:
        short_name: thetao
        timerange: '1950/1952'
        preprocessor: extract_one_region
    scripts: null
```

schlunma reviewed (approved):
Thanks Saskia for making the changes, everything looks good now! The output of the recipe looks as expected :rocket:
